### ISSUE TITLE
Latest version not working

### ISSUE THREAD

**Issue by ground-control-hamburg** [ddoronenko/google-reviews#8]
It seems to be a nice extension, but I can't get it working.
I installed it via composer on a PHP 8.1 / TYPO3 11.5.23 system (ddev), included the api key, the plugin and the places id.
But.. the frontend only shows the else-case "Es wurden keine Bewertungen gefunden!", because the {place} object is empty.
There's no request in the frontend to google.

Meanwhile I tested it on three different systems. Same behaviour. 
Any idea?
Cheers, Sven

**Comment by YoungWorx** [comment]
Hello, I have the same problem, is there a solution? LG Alex

**Comment by ddoronenko** [comment]
@YoungWorx @ground-control-hamburg 
Hello,

I assume, after plugin installation, you have forgotten to include the plugin TypoScript template in the "Template" module -> Info/Modify 
See https://github.com/ddoronenko/google-reviews/blob/master/Documentation/Installation/Index.rst#preparation-include-static-typoscript

See EXT:news example https://docs.typo3.org/p/georgringer/news/11.3/en-us/QuickStart/Configuration.html#quickconfiguration


In case it do not help, provide more info / TYPO3 logs for investigation. Currently, I can't reproduce or confirm the issue.

Cheers, Dima

**Comment by YoungWorx** [comment]
Hello, thank you very much for the message, I integrated the TS in the Tamplate as described. But unfortunately I only get the message “No reviews were found!” I entered the PlaceID correctly in the plugin. Do I still have to apply for a MAPS API from Google? Kind regards, Alex

 

Von: Dima Doronenko ***@***.***> 
Gesendet: Dienstag, 23. Januar 2024 14:14
An: ddoronenko/google-reviews ***@***.***>
Cc: Alexander Möller ***@***.***>; Mention ***@***.***>
Betreff: Re: [ddoronenko/google-reviews] Latest version not working (Issue #8)

 

@YoungWorx <https://github.com/YoungWorx>  @ground-control-hamburg <https://github.com/ground-control-hamburg> 
Hello,

I assume, after plugin installation, you have forgotten to include the plugin TypoScript template in the "Template" module -> Info/Modify
See https://github.com/ddoronenko/google-reviews/blob/master/Documentation/Installation/Index.rst#preparation-include-static-typoscript

See EXT:news example https://docs.typo3.org/p/georgringer/news/11.3/en-us/QuickStart/Configuration.html#quickconfiguration

In case it do not help, provide more info / TYPO3 logs for investigation. Currently, I can't reproduce or confirm the issue.

Cheers, Dima

**Comment by YoungWorx** [comment]
Hello, thank you very much for the message, I integrated the TS in the Tamplate as described. But unfortunately I only get the message “No reviews were found!” I entered the PlaceID correctly in the plugin. Do I still have to apply for a MAPS API from Google? Kind regards, Alex

**Comment by NitinParri** [comment]
If you have an API key with referer restrictions the Places API on won't work with this API key.
The solution is to remove the restrictions.
Check browser console for Google error.

**Comment by Fasna-kv** [comment]
Hello ,

im using the "google_reviews" ( 3.0.0 ) extension for my typo3 site
I am encountering the same issue in my typo3 version 12.4. I have provided the place ID in the field. Please find the attached image, but I am only receiving "No reviews were found!" on the front page (but there are reviews )

"Included static (from extensions)" field.
PLUGIN.TX_TX_GOOGLEREVIEWS_PI1.apiKey = key
Added  Google API key in the "Constant Editor" -> Category 
**PLUGIN.TX_TX_GOOGLEREVIEWS_PI1.apiKey =  **************************
   **TYPO3 Version  12.4.11
  PHP Version  - 8.3.2**

        _<f:if condition="{place.reviews}">_
 this condition is not working 


Thank you.

**Comment by phillued** [comment]
> If you have an API key with referer restrictions the Places API on won't work with this API key. The solution is to remove the restrictions. Check browser console for Google error.

yes, this works for me, too. You cannot have active restrictions of any type for the API key. Why though?

**Comment by felixranesberger** [comment]
Just found the cause: the documentation states that the API key needs to be set under `plugin.tx_googlereviews.apiKey`, but when you look in the code its actually `plugin.tx_googlereviews_pi1.apiKey`. When you use the correct constant name, everything works again. 

There are also other places in the code, where the legacy variable name is used in if casements, but they don't break the functionality at the moment, but should be replaced.

See issue here: https://github.com/ddoronenko/google-reviews?tab=readme-ov-file#installation-short-version

**Comment by ddoronenko** [comment]
@felixranesberger hello,

What version of EXT do you use? It looks like you use TYPO3 v12 + EXT:google_reviews **v3.0** is that right? BUT you follow the documentation of the latest version EXT:google_reviews **4.0** (for TYPO3 v13)

You are right that since EXT:google_reviews 4.0 the constant name was changed from "plugin.tx_googlereviews_pi1." =>  "plugin.tx_googlereviews." Documentation was updated as well.

1) I agree, I need to add a comment in "Troubleshooting / Breaking Changes" documentation about such changes but no changes in the codebase is needed.

2) You need to use the correct version of documentation [according to your EXT version(v3.0)](https://github.com/ddoronenko/google-reviews/blob/v3.0.0/Documentation/Configuration/Index.rst), where you find "plugin.tx_googlereviews_pi1." key.  So, as you can see - you set it correctly

BUT when you update to EXT:google_reviews 4.0 (TYPO3 v13), you will need to update the constant key/path anyway

**Comment by ddoronenko** [comment]
Closing this issue due to no activity.

The troubleshooting section was added with a description of possible issues https://github.com/ddoronenko/google-reviews?tab=readme-ov-file#troubleshooting
